be/src/olap/rowset/segment_v2/segment_writer.cpp

```diff
@@ -186,8 +186,9 @@ Status SegmentWriter::_create_column_writer(uint32_t cid, const TabletColumn& co
 
     // now we create zone map for key columns in AGG_KEYS or all column in UNIQUE_KEYS or DUP_KEYS
     // except for columns whose type don't support zone map.
-    opts.need_zone_map = column.is_key() || schema->keys_type() != KeysType::AGG_KEYS;
-    opts.need_bloom_filter = column.is_bf_column();
+    opts.need_zone_map = (column.is_key() || schema->keys_type() != KeysType::AGG_KEYS) &&
```

Comment:
what about other types like jsonb or array map? do they need_zone_map?why is variant special here

Reply:
so why variant is special here?

Reply:
Line 221 will not check the variant column.

Reply:
add check for variant